### PR TITLE
[MP-2175] React Native Change js to tsx

### PR DESCRIPTION
**Pull request by lewis-rokt**
### Background ###

This PR makes multiple changes to modernise the react native SDK, primarily by changing some files to TypeScript and adding linting to ensure that the distributed files are correctly formatted in response to a syntax issue raised.

Fixes [MP-2175](https://rokt.atlassian.net/browse/MP-2175?atlOrigin=eyJpIjoiY2QxMTVmMTkwYTQ4NGIyZmJhN2VmMzExMDAwZGQ3YzAiLCJwIjoiaiJ9)

#### Rokt.Widget
- Changed `rokt-embedded-view.android` and `rokt-embedded-view.ios` to typescript.
- Made multiple changes as enforced by typescript (adding types, changing how state is updated, changing to arrow functions to avoid the need for binding).
- Added `typescript-eslint` to lint relevant source files.
- Moved .tsx and index.js files to `src` directory.
- Added a build step to transpile the .tsx to .js.
- The resulting package uses the js generated by tsc, ensuring that there are no syntax issues

#### RoktSampleApp
- Applied prettier formatting to .js files.
- Fixed a couple of minor bugs

### How Has This Been Tested? ###

Verified both sample apps using the local package containing the generated js

### Notes

react-native can use .tsx files out-of-box as babel transpiles the files to js as part of the bundling process. However, for the most compatibility (custom bundling processes, random parsing throwing errors at .tsx files, etc.) I kept the output files as .js by adding the tsc building step (even modules created specifically for react-native use js instead of ts from what I found). The only difference to the package used by partners is the file structure so there shouldn't be any compatibility issues.

Build failing is because I didn't update the CircleCI pipeline, buildkite works fine.

### Checklist: ###

- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have verified that CI completes successfully.
- [x] All insignificant commits have been squashed.

### CODE REVIEW

**Review by sahil-rokt — APPROVED** [review]

**Review by thomson-t — APPROVED** [review]

**Comment by danial-rokt** [comment]
Do you think it is a good idea to test it by releasing an alpha version? To make sure there is no incompatibility issue.

File: Rokt.Widget/package.json [hunk 1]
```diff
@@ -1,9 +1,11 @@
 {
   "name": "@rokt/react-native-sdk",
-  "version": "3.13.0",
+  "version": "3.14.0-alpha.0",
   "description": "Rokt Mobile SDK to integrate ROKT Api into ReactNative application",
-  "main": "index.js",
+  "main": "dist/index.js",
```
Comment (thomson-t) [inline Rokt.Widget/package.json:5]: I think it is better to add the `types` attribute as well, https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package

Comment (lewis-rokt) [reply Rokt.Widget/package.json:5]: Sure, done

**Review by thomson-t — APPROVED** [review]

**Comment by lewis-rokt** [comment]
> Do you think it is a good idea to test it by releasing an alpha version? To make sure there is no incompatibility issue.

Tested with alpha version in the demo app and it worked for both platforms. VSCode also picked up the types properly.

**Review by danial-rokt — APPROVED** [review]

**Comment by danial-rokt** [comment]
> > Do you think it is a good idea to test it by releasing an alpha version? To make sure there is no incompatibility issue.
> 
> Tested with alpha version in the demo app and it worked for both platforms. VSCode also picked up the types properly.

Awesome thanks

**Pull request merged**